src/nvim/os/env.c

```diff
@@ -429,24 +433,36 @@ void expand_env_esc(char_u *restrict srcp,
         } else // NOLINT
 #endif
         {
-          while (c-- > 0 && *tail != NUL && vim_isIDc(*tail)) {
+          while (c-- > 0 && *tail != NUL && ((vim_isIDc(*tail))
+#ifdef WIN32
+                                             || (*src == '%' && *tail != '%')
+#endif
+                                             )) {  //  NOLINT(whitespace/parens)
             *var++ = *tail++;
           }
         }
 
-#if defined(UNIX)
+#if defined(UNIX) || defined(WIN32)
+# ifdef UNIX
         // Verify that we have found the end of a Unix ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
+# else
+        if (*src == '%' && *tail != '%') {
```

Comment:
we could define these platform-specific start/end , though I guess it doesn't help much :

```c
#if defined(UNIX)
# define ENV_START = '%';
# define ENV_END = '%';
#else
# define ENV_START = '%';
# define ENV_END = '%';
#endif
```

Reply:
It seemed like a good idea, but when I examined it, I found that we could not reduce the `#ifdef`. It is rather hard to see, so it seems better to quit.